### PR TITLE
Add proper tag for videos in playlists

### PR DESCRIPTION
**Pull request by VioletRed**
Add a videotag for video files in M3U and PLS files.

### CODE REVIEW

**Comment by da-anda** [comment]
can you explain what this change fixes or is there a related trac ticket where we can read about the issue?

**Comment by VioletRed** [comment]
Sorry I forgot to link the proper discussion.

I submitted a patch to fix SmartPlaylists, that were skipping music videos even if "mixed" was selected as type. I posted about [it here](http://forum.kodi.tv/showthread.php?tid=216425). 

The user that was helping me with testing complained that [videos got skipped after a song is played](http://forum.kodi.tv/showthread.php?tid=216425&pid=1932708#pid1932708), it turned out to be a problem with [M3U playlists](http://forum.kodi.tv/showthread.php?tid=216425&pid=1934743#pid1934743) that seems easy enough to fix to give it a try.

Basically, when creating a playlists from a M3U file, all items are tagged as "songs" what prevents PlayerCoreFactory to determine the proper core to open a file. Specifically, PAPlayer is used to play video files, and fails.

**Comment by VioletRed** [comment]
@da-anda Is the improvement useful or should I close the pull request?

**Comment by da-anda** [comment]
@VioletRed sounds good to me, but a dev with knowledge in this area needs to have a look at this

**Comment by Montellese** [comment]
Are you calling `Reset()` just because you want to force `CFileItem` to create a new `CVideoInfoTag` instance?

**Comment by VioletRed** [comment]
@Montellese Yes, and I guess by your comment that that's not the proper way.

**Comment by Montellese** [comment]
@VioletRed: It's not wrong, but it's not obvious. Maybe simply add a comment explaining that there is no videoinfotag and that this add's an empty one. Maybe also add a check for `HasVideoInfoTag()` before overwriting it.

**Comment by VioletRed** [comment]
@Montellese Fixed.

**Comment by Montellese** [comment]
Thanks.
jenkins build and merge

**Comment by VioletRed** [comment]
It seems like "mirrors.xbmc.org" is down right now:

```
Scanning dependencies of target ogg
[  1%] Creating directories for 'ogg'
[  2%] Performing download step (download, verify and extract) for 'ogg'
-- downloading...
     src='http://mirrors.xbmc.org/build-deps/sources/libogg-1.3.0.tar.gz'
     dst='C:/jenkins/slave/workspace/WIN-32/project/cmake/addons/build/download/libogg-1.3.0.tar.gz'
     timeout='none'
CMake Error at ogg-stamp/download-ogg.cmake:22 (message):
  error: downloading
  'http://mirrors.xbmc.org/build-deps/sources/libogg-1.3.0.tar.gz' failed

    status_code: 7
    status_string: "couldn't connect to server"
    log: About to connect() to mirrors.xbmc.org port 80 (#0)
    Trying 89.16.176.16... Timed out```

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by Montellese** [comment]
OSX32 build failure is unrelated.
